test/unified_memory_allocation/memoryPool.hpp

```diff
@@ -73,6 +76,56 @@ TEST_P(umaPoolTest, pow2AlignedAlloc) {
     }
 }
 
+TEST_P(umaPoolTest, allocZeroSize) {
+    static constexpr size_t allocSize = 0;
+    auto *ptr = umaPoolMalloc(pool.get(), allocSize);
+    umaPoolFree(pool.get(), ptr);
+}
+
+TEST_P(umaPoolTest, freeNullptr) {
+    void *ptr = nullptr;
+    umaPoolFree(pool.get(), ptr);
+    ASSERT_EQ(ptr, nullptr);
```

Comment:
Can this ever be false? umaPoolFree cannot modify the ptr.

Reply:
It is removed now.